test/e2e/util/util.go

```diff
@@ -1252,3 +1255,54 @@ func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx conte
 		}
 	})
 }
+
+func EnsureGuestWebhooksValidated(t *testing.T, ctx context.Context, guestClient crclient.Client) {
+	t.Run("EnsureGuestWebhooksValidated", func(t *testing.T) {
+		guestWebhookConf := &admissionregistrationv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-webhook",
+				Namespace: "default",
+				Annotations: map[string]string{
+					"service.beta.openshift.io/inject-cabundle": "true",
+				},
+			},
+		}
+
+		sideEffectsNone := admissionregistrationv1.SideEffectClassNone
+		guestWebhookConf.Webhooks = []admissionregistrationv1.ValidatingWebhook{{
+			AdmissionReviewVersions: []string{"v1"},
+			Name:                    "etcd-client.example.com",
+			ClientConfig: admissionregistrationv1.WebhookClientConfig{
+				URL: pointer.String("https://etcd-client:2379"),
+			},
+			Rules: []admissionregistrationv1.RuleWithOperations{{
+				Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"pods"},
+				},
+			}},
+			SideEffects: &sideEffectsNone,
+		}}
+
+		if err := guestClient.Create(ctx, guestWebhookConf); err != nil {
+			t.Errorf("failed to create webhook: %v", err)
+		}
+
+		err := wait.PollImmediateWithContext(ctx, 5*time.Second, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
```

Comment:
wouldn't it be a bug if this takes for than a few secs? why 5min?

Reply:
right, I just didn't want to have flakes in CI because of this.
I changed it to 1min